src/services/firebase.test.ts

```diff
@@ -112,6 +112,47 @@ it('fetch match list from firebase', done => {
   });
 });
 
+it('Should update the phone numbers', done => {
+  // write something to gameportal/phoneNumberToUserId
+  const testPhoneNumber: string = '123456789';
+  // Since our test use anonymous login
+  // and the rules only allow you to write there if you have auth.token.phone_number
+  // we can not add in gamePortal/PhoneNumberToUserId/${phoneNumber}
+  // So firebase rules add "123456789" for test
+  const phoneNumberFbr: fbr.PhoneNumber = {
+    userId: ourFirebase.getUserId(),
+    timestamp: ourFirebase.getTimestamp()
+  };
+  ourFirebase.refSet(
+    ourFirebase.getRef(`/gamePortal/phoneNumberToUserId/${testPhoneNumber}`),
+    phoneNumberFbr
+  );
+  // get string from contact and convert them to string
+  const phoneNumbers: string[] = [];
+  phoneNumbers.push(testPhoneNumber);
+  ourFirebase.updateUserIdsAndPhoneNumbers(phoneNumbers);
+  // check if store has been updated
+  store.subscribe(() => {
+    const newMap = store.getState().userIdsAndPhoneNumbers;
+    console.log('store:' + store.getState().userIdsAndPhoneNumbers);
+    const phoneNumberToUserId = newMap['phoneNumberToUserId'];
+    if (phoneNumberToUserId[testPhoneNumber]) {
```

Comment:
Seems like you are missing some code here.

Reply:
What I want to do in if block is to check if phoneNumberToUserId has key testPhoneNumber or not. I google it and find this way.